app/views/artisans/StudentSolutionsView.coffee

```diff
@@ -0,0 +1,394 @@
+    @languages = @$('#languageSelect').val()
+    @levelSlug = @$('#levelSlug').val()
+    @startFetchingData()
+
+  onLevelLoaded: (level) =>
+    @level = level
+    @resetLevelInfo()
+
+    for solution in level.getSolutions()
+      continue unless solution.source and solution.language in @validLanguages
+      ast = @parseSource solution.source, solution.language
+      continue unless ast
+      ast = @walkASTCorrect ast, @processASTNode
+      hash = @sha1(JSON.stringify(ast))
+      @intended[solution.language] = hash: hash, source: solution.source
+
+    defaults = @getDefaultCode level
+    for language, source of @getDefaultCode level
+      continue unless source and language in @validLanguages
+      ast = @parseSource source, language
+      continue unless ast
+      ast = @walkASTCorrect ast, @processASTNode
+      hash = @sha1(JSON.stringify(ast))
+      @defaultcode[language] = hash: hash, source: source
+    @fetchSessions()
+
+  getDefaultCode: (level) ->
+    # TODO: put this into Level? if so, also use it in TeacherCourseSolutionView
+    heroPlaceholder = level.get('thangs').filter((x) => x.id == 'Hero Placeholder').pop()
+    comp = heroPlaceholder?.components.filter((x) => x.original.toString() == '524b7b5a7fc0f6d51900000e' ).pop()
+    programmableMethod = comp?.config.programmableMethods.plan
+    result = {}
+
+    parseTemplate = (src, context) =>
+      try
+        res = _.template(src)(context)
+        return res
+      catch e
+        console.warn "Template Error"
+        console.log src
+        return src
+
+    # javascript
+    if programmableMethod.source
+      src = programmableMethod.source
+      src = parseTemplate(src, programmableMethod.context)
+      result['javascript'] = src
+    # non-javascript
+    for key in _.keys(programmableMethod.languages)
+      continue unless key in ['python']
+      src = programmableMethod.languages[key]
+      src = parseTemplate(src, programmableMethod.context)
+      result[key] = src
+
+    result
+
+  parseSource: (src, lang) =>
+    if lang is 'python'
+      aether = new Aether language: 'python'
+      tsrc = aether.transpile(src)
+      ast = aether.ast
+      # TODO: continue if error
+      # aether.problems?
+    if lang is 'javascript'
+      try
+        ast = parser(src)
+      catch e
+        @stats[lang].errors += 1
+        return null
+    ast
+
+  # TODO: expose walkASTCorrect from aether instead of recreating it here?
+  walkASTCorrect: (node, fn) ->
+    for key, child of node
+      if _.isArray child
+        for grandchild in child
+          if _.isString grandchild?.type
+            @walkASTCorrect grandchild, fn
+      else if _.isString child?.type
+        @walkASTCorrect child, fn
+    fn node
+
+  processASTNode: (node) =>
+    return unless node?
+    delete node.range if node.range
+    delete node.loc if node.loc
+    delete node.originalRange if node.originalRange
+    node
+
+  # TODO: replace this with a library if possible
```

Comment:
Yep, someone should definitely do that, could be one of us if you're not familiar with installing bower stuff. Preferably a minimal package so as not to inflate our dependencies hugely (eg. just `sha1` rather than a whole `crypto` thing). This might do that: https://www.npmjs.com/package/crypto-js